### PR TITLE
(FACT-2176) Return structured data on executable facts

### PR DESCRIPTION
**Pull request by florindragos**

### CODE REVIEW

File: lib/custom_facts/util/parser.rb [hunk 1]
```diff
@@ -154,7 +165,7 @@ def parse_results
           shell_command =
             "\"#{powershell}\" -NoProfile -NonInteractive -NoLogo -ExecutionPolicy Bypass -File \"#{filename}\""
           output = LegacyFacter::Core::Execution.exec(shell_command)
-          KeyValuePairOutputFormat.parse(output)
+          parse_executable_output output
```
Comment (Filipovici-Andrei) [inline lib/custom_facts/util/parser.rb]: The parameters need to be within parentheses.

File: lib/custom_facts/util/parser.rb [hunk 2]
```diff
@@ -120,7 +131,7 @@ def parse_results
 
       class ScriptParser < Base
         def parse_results
-          KeyValuePairOutputFormat.parse LegacyFacter::Core::Execution.exec(quote(filename))
+          parse_executable_output LegacyFacter::Core::Execution.exec(quote(filename))
```
Comment (Filipovici-Andrei) [inline lib/custom_facts/util/parser.rb]: The parameters need to be within parentheses.

**Review by Filipovici-Andrei — CHANGES_REQUESTED** [review]

File: lib/custom_facts/util/parser.rb [hunk 3]
```diff
@@ -65,6 +65,17 @@ def results
...
+        def parse_executable_output(output)
+          res = nil
+          begin
+            res = YAML.safe_load output
+          rescue Exception => e
+            Facter.debug("Could not parse executable fact output as YAML or JSON (#{e.message})")
+          end
+          res = KeyValuePairOutputFormat.parse output unless res.is_a?(Hash)
+          res
+        end
```
Comment (oanatmaria) [inline lib/custom_facts/util/parser.rb]: Could you please add a test for these changes?

Comment (florindragos) [reply lib/custom_facts/util/parser.rb]: done

**Review by oanatmaria — APPROVED** [review]

**Review by Filipovici-Andrei — APPROVED** [review]

**Pull request merged**